keps/sig-apps/4368-support-managed-by-for-batch-jobs/README.md

```diff
@@ -362,6 +363,52 @@ We may come up with more validation rules during the implementation phase.
 The API comments to the Job status API fields will be updated to make the contract
 clear.
 
+We also plan to add a validation rule to make sure that the terminal conditions
+(`Failed` and `Complete`) are only added to Job when all pods are terminated.
+For that we plan to follow the approach described [below](#terminating-pods-and-terminal-job-conditions).
+
+#### Terminating pods and terminal Job conditions
+
+During the development process of Alpha in 1.30 we considered adding a
+validation rule enforcing that the Job terminal conditions (`Failed` or `Complete`)
+are only added when all pods are terminated (`status.terminating=0` and
+`status.ready=0`). However, the rule turned out to be violated by the built-in
+Job controller (see issue [#123775](https://github.com/kubernetes/kubernetes/issues/123775)).
+
+We are going to solve this issue by delaying the addition of the Job terminal
+conditions (Failed or Complete) until the pods are terminated
+(`terminating=0` and `ready=0`).
+
+One complication of this approach is that pod termination may take an
+arbitrarily long if the non-standard pod graceful termination period
+(`terminationGracePeriodSeconds`) is configured (30s by default). In order to
+give the API clients flexibility to know the fate of a Job as soon as possible
+we extend the scope for the following Job conditions:
+- `FailureTarget` - introduced in [Pod Failure Policy KEP](https://github.com/kubernetes/enhancements/issues/3329)
+- `SuccessCriteriaMet` - introduced in [Success Policy KEP](https://github.com/kubernetes/enhancements/issues/3998)
```

Comment:
Again, I prefer links to KEPs themselves, rather than the issue. Less clicking through :wink:

Reply:
Sure, done